atomicapp/applogging.py

```diff
@@ -0,0 +1,153 @@
+
+ Atomic App is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public License
+ along with Atomic App. If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import sys
+import logging
+
+from atomicapp.constants import (LOGGER_COCKPIT,
+                                 LOGGER_DEFAULT)
+
+
+class colorizeOutputFormatter(logging.Formatter):
+    """
+    A class to colorize the log msgs based on log level
+    """
+
+    def format(self, record):
+        # Call the parent class to do formatting.
+        msg = super(colorizeOutputFormatter, self).format(record)
+
+        # Now post process and colorize if needed
+        if record.levelno == logging.DEBUG:
+            msg = self._colorize(msg, 'cyan')
+        elif record.levelno == logging.WARNING:
+            msg = self._colorize(msg, 'yellow')
+        elif record.levelno == logging.INFO:
+            msg = self._colorize(msg, 'white')
+        elif record.levelno == logging.ERROR:
+            msg = self._colorize(msg, 'red')
+        else:
+            raise Exception("Invalid logging level {}".format(record.levelno))
+        return self._make_unicode(msg)
+
+    def _colorize(self, text, color):
+        """
+        Colorize based upon the color codes indicated.
+        """
+        # Console colour codes
+        colorCodes = {
+            'white': '0', 'bright white': '1;37',
+            'blue': '0;34', 'bright blue': '1;34',
+            'green': '0;32', 'bright green': '1;32',
+            'cyan': '0;36', 'bright cyan': '1;36',
+            'red': '0;31', 'bright red': '1;31',
+            'purple': '0;35', 'bright purple': '1;35',
+            'yellow': '0;33', 'bright yellow': '1;33',
+        }
+        return "\033[" + colorCodes[color] + "m" + text + "\033[0m"
+
+    def _make_unicode(self, input):
+        """
+        Convert all input to utf-8 for multi language support
+        """
+        if type(input) != unicode:
+            input = input.decode('utf-8')
+        return input
+
+
+class Logging:
+
+    @staticmethod
+    def setup_logging(verbose=None, quiet=None, outputtype=None):
+        """
+        This function sets up logging based on the outputtype requested.
+        The 'none' level outputs not logs at all
+        The 'cockpit' level outputs just logs for the cockpit logger
+        The 'stdout' level prints out normal log msgs (no cockpit) without color
+        The 'default' level prints out normal log msgs (no cockpit) with color
+        """
+
+        # If no outputtype was set then let's have a sane default
+        # If connected to a tty, then default to color, else, no color
+        if not outputtype:
+            if sys.stdout.isatty():
+                outputtype = 'stdoutcolor'
```

Comment:
I wonder if we should use something shorter for the cli that would make sense for a variable name as well. would --logtype on the cli work? then here we could use logtype as the var name.